error.go

```diff
@@ -0,0 +1,37 @@
+package zapdriver
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap/zapcore"
+)
+
+func ErrFormat(err error) string {
+	if err == nil {
+		return ""
+	}
+	type stackTracer interface {
+		StackTrace() errors.StackTrace
+	}
+	cause := errors.Cause(err)
+	if stackTrace, ok := cause.(stackTracer); ok {
```

Comment:
If I read up on https://pkg.go.dev/github.com/pkg/errors#Cause then it seems weird to use `errors.Cause` before taking `.(stackTracer)` because maybe the upper/last error also contained a stacktrace. Do `stackTracer` implementations include the stack frames from the Cause too?

Can we somehow come closer to this layout (note the `Caused by`)?

```
Exception in thread "main" com.myproject.module.MyProjectFooBarException: The number of FooBars cannot be zero
    at com.myproject.module.MyProject.anotherMethod(MyProject.java:19)
    at com.myproject.module.MyProject.someMethod(MyProject.java:12)
    at com.myproject.module.MyProject.main(MyProject.java:8)
Caused by: java.lang.ArithmeticException: The denominator must not be zero
    at org.apache.commons.lang3.math.Fraction.getFraction(Fraction.java:143)
    at com.myproject.module.MyProject.anotherMethod(MyProject.java:17)
    ... 2 more
```